add-on/src/lib/redirect-handler/blockOrObserve.ts

```diff
@@ -24,7 +24,11 @@ interface messageToSelf {
 // We need to check if the browser supports the declarativeNetRequest API.
 // TODO: replace with check for `Blocking` in `chrome.webRequest.OnBeforeRequestOptions`
 // which is currently a bug https://bugs.chromium.org/p/chromium/issues/detail?id=1427952
-export const supportsBlock = !(browser.declarativeNetRequest?.MAX_NUMBER_OF_DYNAMIC_AND_SESSION_RULES === 5000)
+// this needs to be a function call, because in tests we mock browser.declarativeNetRequest
+// the way sinon ends up stubbing it, it's not directly available in the global scope on import
+// rather it gets replaced dynamically when the module is imported. Which means, we can't
+// just check for the existence of the property, we need to call the browser instance at that point.
+export const supportsBlock = (): boolean => !(browser.declarativeNetRequest?.MAX_NUMBER_OF_DYNAMIC_AND_SESSION_RULES === 5000)
```

Comment:
nit: this check will break when Google changes limit.

Maybe:

```suggestion
export const noDeclarativeNetRequest = (): boolean => !(typeof browser.declarativeNetRequest === 'function')
```

?

Reply:
oof, given [Google lost interest in MV3](https://groups.google.com/u/0/a/chromium.org/g/chromium-extensions/c/zQ77HkGmK9E) that is not getting fixed any time soon.

Maybe at least remove dependency on specific number?

```suggestion
export const supportsBlock = (): boolean => !(browser.declarativeNetRequest?.MAX_NUMBER_OF_DYNAMIC_AND_SESSION_RULES > 0)
```